starlette_discord/client.py

```diff
@@ -13,42 +15,54 @@ class DiscordOAuthSession(OAuth2Session):
     ----------
     code: :class:`str`
         Authorization code included with user request after redirect from Discord.
+    token: :class:`Union[str, int, float]`
+        A previously generated, valid, access token to use instead of the OAuth code exchange
     client_id: :class:`int`
         Your Discord application client ID.
     scope: :class:`str`
         Discord authorization scopes separated by %20.
     redirect_uri: :class:`str`
         Your Discord application redirect URI.
     """
-    def __init__(self, code, client_id, client_secret, scope, redirect_uri):
+
+    def __init__(self, client_id, client_secret, scope, redirect_uri,
+                 code: str = None,  # obtained during login callback
+                 token: Dict[str, Union[str, int, float]] = None  # example {'access_token': 'sffd8fds90ds890fs'}
+                 ):
+
+        if not code and not token:
+            raise ValueError('Either code or token are required to construct the session')
+        elif code and token:
+            raise ValueError('Use either :param code: or :param token:, not both')
+        elif token:
+            if not token.get('access_token'):
+                raise ValueError(':param token: requires "access_token" key but is missing.')
+            elif not token.get('token_type'):  # this is not required for the discord class but for the parent class
+                token['token_type'] = 'Bearer'
+
         self._discord_auth_code = code
         self._discord_client_secret = client_secret
-        self._discord_token = None
+        self.discord_token = token
```

Comment:
this would probably be better off as a property.

Reply:
agreed